WalletWasabi.Fluent/Behaviors/HideShowBehavior.cs

```diff
@@ -0,0 +1,62 @@
+using System;
+using System.ComponentModel;
+using System.Reactive.Disposables;
+using System.Reactive.Linq;
+using System.Runtime.InteropServices;
+using Avalonia.Controls;
+using ReactiveUI;
+using WalletWasabi.Services;
+
+namespace WalletWasabi.Fluent.Behaviors
+{
+	public class HideShowBehavior : DisposingBehavior<Window>
+	{
+		private const string icon_path = "/Assets/WasabiLogo.ico";
+
+		protected override void OnAttached(CompositeDisposable disposables)
+		{
+			if (AssociatedObject is null)
+			{
+				return;
+			}
+
+			// On macOs the Hide/Show is a natural feature.
+			if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+			{
+				Observable
+					.FromEventPattern(Services.SingleInstanceChecker,
+						nameof(SingleInstanceChecker.OtherInstanceStarted))
+					.ObserveOn(RxApp.MainThreadScheduler)
+					.Subscribe(_ =>
+					{
+						if (AssociatedObject.WindowState == WindowState.Minimized)
+						{
+							AssociatedObject.WindowState = (WindowState)Enum.Parse(typeof(WindowState), Services.UiConfig.WindowState);
+						}
+
+						// Fixes: https://github.com/zkSNACKs/WalletWasabi/issues/6309
+						var temp = AssociatedObject.WindowState;
+						AssociatedObject.Show();
+						AssociatedObject.WindowState = temp;
+
+						AssociatedObject.BringIntoView();
+					})
+					.DisposeWith(disposables);
+			}
+
+			//TODO: we need the close button click only, external close request should not be cancelled.
+			Observable
+				.FromEventPattern<CancelEventArgs>(AssociatedObject, nameof(AssociatedObject.Closing))
+				.ObserveOn(RxApp.MainThreadScheduler)
+				.Subscribe(args =>
+				{
+					if (Services.UiConfig.HideOnClose)
+					{
+						//args.EventArgs.Cancel = true;
+						//AssociatedObject.Hide();
+					}
+				})
+				.DisposeWith(disposables);
```

Comment:
I have tried to comment out this section and it still works for me as I would expect. Could you please clarify why it is so? Or what this code should do?